src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -1957,6 +1959,27 @@ public Task inspectTask(final String taskId) throws DockerException, Interrupted
   }
 
   @Override
+  public NodeInfo inspectNode(final String nodeId) throws DockerException, InterruptedException {
+    assertApiVersionIsAbove("1.24");
+
+    WebTarget resource = resource().path("nodes")
+        .path(nodeId);
+
+    try {
+      return request(GET, NodeInfo.class, resource, resource.request(APPLICATION_JSON_TYPE));
+    } catch (DockerRequestException e) {
+      switch (e.status()) {
+        case 404:
+          throw new NodeNotFoundException(nodeId);
+        case 503:
```

Comment:
Looks like its not being used in the newer versions for some other purpose.
Would you like me to modify this to throw the NonSwarmNodeException for the 406?